src/SyntaxHighlighter/Cpp.cpp

```diff
@@ -43,8 +46,8 @@ const char* const kOpenCommentRegex = "\\/\\*([^\\*]|[\\*]+[^\\/])*?$";
 const char* const kEndCommentRegex = "[\\s\\S]*\\*\\/";
 // Match with a line without a closing comment
 const char* const kNoEndCommentRegex = "([^\\*]|\\*+[^\\/\\*])*$";
-// A word who is followed with an open-parenthesis (
-const char* const kFunctionRegex = "(?<=[^\\(\\:\\.\\w\\>])[a-z_]\\w*(?=\\s*\\()";
+// int Function( or Namespace::FunctionName patterns
+const char* const kFunctionDefinitionRegex = "(?<=\\w)\\s+(\\w*::)*\\w*(?=\\()";
```

Comment:
so you don't allow `_` in function names? Remember Google Style Guide as `snake_case()`/`set_snake_case(T t)` for simple accessor methods. If that is not easily doable, also feel free to ignore.

Reply:
`\w` includes a-z, A-Z, 0-9 ,and "_".

Reply:
The last `\w*` should probably be `\w+`, no? And the one before `::` as well, isn't it?

Reply:
\w includes '_' :)
You can see 2 examples there in the screenshots with functions having '_'.

I'm differentiating them using a horrible heuristic here. I'm asking for a character before the function(

In that way:
"int function();'' has the 't' of int
and "lets_execute_function()" not.

"(?<=\\w)\\s+" is doing that.
?<= means looking behind without painting it
and \\s+ is the spaces between that character and the function itself.

BTW, thanks for your comment, I've just realized that I have to polish this a little bit to avoid having number as a first character here. :)

Reply:
Yes, thanks Henning

Reply:
Changed: "(?<=\\w)\\s+([A-Za-z_]\\w*::)*[A-Za-z_]\\w*(?=\\()"

Instead of **_\w+_**, I used **_[A-Za-z_]\w*_**, because it shouldn't start with a number